knimin/lib/data_access.py

```diff
@@ -3952,6 +3952,70 @@ def prepare_targeted_libraries(self, plate_links, email, robot, tm300tool,
             # list into the desired output: [plate_id, plate_id, ...]
             return list(chain.from_iterable(chain.from_iterable(res)))
 
+    def quantify_targeted_plate(self, plate_id, data, vals):
+        """Adds quantification information to the target plate
+
+        Parameters
+        ----------
+        plate_id : int
+            The target plate id
+        data : str
+            Which information to add, it can be 'raw_concentration' or
+            'mod_concentration'.
+        vals : 2d numpy array of floats
+            The per-well DNA concentration
+
+        Raises
+        ------
+        ValueError
+            If data is not one of the valid values, see Parameters
+            If vals shape is not the same than the plate
+
+        Notes
+        -----
+            You always will need to first add raw_concentration as it's an
+            insert and mod_concentration, which is an update
+        """
+        valid_data = ['raw_concentration', 'mod_concentration']
+        if data not in valid_data:
+            raise ValueError(
+                "Not a valid data: %s, should be: %s" % (data, valid_data))
+
+        tp = self.read_targeted_plate(plate_id)
```

Comment:
Minor but can you move this block inside the `TRN` block? They're accessing the DB so it makes sense to be inside the transaction.